src/test/scala/org/bitcoins/core/crypto/TransactionSignatureCreatorSpec.scala

```diff
@@ -37,16 +38,62 @@ class TransactionSignatureCreatorSpec extends Properties("TransactionSignatureCr
         val program = ScriptProgram(txSignatureComponent)
 
         val result = ScriptInterpreter.run(program)
+
         result == ScriptOk
   }
 
   property("generate a valid signature for a p2sh transaction") =
-    Prop.forAll(TransactionGenerators.p2SHTransaction) {
+    Prop.forAll(TransactionGenerators.signedP2SHTransaction) {
       case (txSignatureComponent: TransactionSignatureComponent, _) =>
         //run it through the interpreter
         val program = ScriptProgram(txSignatureComponent)
         val result = ScriptInterpreter.run(program)
         //can be ScriptErrorPushSize if the redeemScript is larger than 520 bytes
         Seq(ScriptOk, ScriptErrorPushSize).contains(result)
     }
+
+  property("generate a valid signature for a valid and spendable cltv transaction") =
+    Prop.forAllNoShrink(TransactionGenerators.spendableCLTVTransaction :| "cltv_spendable") {
+      case (txSignatureComponent: TransactionSignatureComponent, _, scriptNumber) =>
+        //run it through the interpreter
+        require(txSignatureComponent.transaction.lockTime.underlying >= scriptNumber.underlying, "TxLocktime must be satisifed so it should be greater than or equal to  " +
+          "the cltv value. Got TxLockTime : " + txSignatureComponent.transaction.lockTime.underlying + " , and cltv Value: " +
+          scriptNumber.underlying)
+        val program = ScriptProgram(txSignatureComponent)
+        val result = ScriptInterpreter.run(program)
+        val cltv = CLTVScriptPubKey(txSignatureComponent.scriptPubKey.hex)
+        Seq(ScriptOk).contains(result)
+    }
+
+  property("generate a valid signature for a validly constructed, but NOT spendable cltv transaction") =
```

Comment:
Since we don't check the validity of the signature, we probably shouldn't put it in the name of the test case. Should be something like `fail to verify a transaction whose locktime has not passed yet`